packages/eslint-plugin-svelte/src/rules/prefer-svelte-reactivity.ts

```diff
@@ -0,0 +1,243 @@
+				for (const { node, path } of referenceTracker.iterateGlobalReferences({
+					Date: {
+						[ReferenceTracker.CONSTRUCT]: true
+					},
+					Map: {
+						[ReferenceTracker.CONSTRUCT]: true
+					},
+					Set: {
+						[ReferenceTracker.CONSTRUCT]: true
+					},
+					URL: {
+						[ReferenceTracker.CONSTRUCT]: true
+					},
+					URLSearchParams: {
+						[ReferenceTracker.CONSTRUCT]: true
+					}
+				})) {
+					if (path[0] === 'Date' && isDateMutable(referenceTracker, node)) {
+						context.report({
+							messageId: 'mutableDateUsed',
+							node
+						});
+					}
+					if (path[0] === 'Map' && isMapMutable(referenceTracker, node)) {
+						context.report({
+							messageId: 'mutableMapUsed',
+							node
+						});
+					}
+					if (path[0] === 'Set' && isSetMutable(referenceTracker, node)) {
+						context.report({
+							messageId: 'mutableSetUsed',
+							node
+						});
+					}
+					if (path[0] === 'URL' && isURLMutable(referenceTracker, node)) {
+						context.report({
+							messageId: 'mutableURLUsed',
+							node
+						});
+					}
+					if (path[0] === 'URLSearchParams' && isURLSearchParamsMutable(referenceTracker, node)) {
+						context.report({
+							messageId: 'mutableURLSearchParamsUsed',
+							node
+						});
+					}
+				}
+			}
+		};
+	}
+});
+
+function isDateMutable(referenceTracker: ReferenceTracker, ctorNode: Expression): boolean {
+	return (
+		Array.from(
+			referenceTracker.iteratePropertyReferences(ctorNode, {
+				setDate: {
+					[ReferenceTracker.CALL]: true
+				},
+				setFullYear: {
+					[ReferenceTracker.CALL]: true
+				},
+				setHours: {
+					[ReferenceTracker.CALL]: true
+				},
+				setMilliseconds: {
+					[ReferenceTracker.CALL]: true
+				},
+				setMinutes: {
+					[ReferenceTracker.CALL]: true
+				},
+				setMonth: {
+					[ReferenceTracker.CALL]: true
+				},
+				setSeconds: {
+					[ReferenceTracker.CALL]: true
+				},
+				setTime: {
+					[ReferenceTracker.CALL]: true
+				},
+				setUTCDate: {
+					[ReferenceTracker.CALL]: true
+				},
+				setUTCFullYear: {
+					[ReferenceTracker.CALL]: true
+				},
+				setUTCHours: {
+					[ReferenceTracker.CALL]: true
+				},
+				setUTCMilliseconds: {
+					[ReferenceTracker.CALL]: true
+				},
+				setUTCMinutes: {
+					[ReferenceTracker.CALL]: true
+				},
+				setUTCMonth: {
+					[ReferenceTracker.CALL]: true
+				},
+				setUTCSeconds: {
+					[ReferenceTracker.CALL]: true
+				},
+				setYear: {
+					[ReferenceTracker.CALL]: true
+				}
+			})
+		).length > 0
+	);
+}
+
+function isMapMutable(referenceTracker: ReferenceTracker, ctorNode: Expression): boolean {
+	return (
+		Array.from(
+			referenceTracker.iteratePropertyReferences(ctorNode, {
+				clear: {
+					[ReferenceTracker.CALL]: true
+				},
+				delete: {
+					[ReferenceTracker.CALL]: true
+				},
+				set: {
+					[ReferenceTracker.CALL]: true
+				}
+			})
+		).length > 0
```

Comment:
I think it can be written as follows:

```suggestion
	    !referenceTracker.iteratePropertyReferences(ctorNode, {
		    clear: {
			    [ReferenceTracker.CALL]: true
		    },
		    delete: {
			    [ReferenceTracker.CALL]: true
		    },
		    set: {
			    [ReferenceTracker.CALL]: true
		    }
	    }).next().done
```

Reply:
Great, I was struggling a little bit with iterators :D